tests/Test.Microsoft.Identity.LabInfrastructure/LabResponse.cs

```diff
@@ -1,70 +1,58 @@
-﻿//-----------------------------------------------------------------------------
-//
-// Copyright (c) Microsoft Corporation.
-// All rights reserved.
-//
-// This code is licensed under the MIT License.
-//
-// Permission is hereby granted, free of charge, to any person obtaining a copy
-// of this software and associated documentation files(the "Software"), to deal
-// in the Software without restriction, including without limitation the rights
-// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
-// copies of the Software, and to permit persons to whom the Software is
-// furnished to do so, subject to the following conditions :
-//
-// The above copyright notice and this permission notice shall be included in
-// all copies or substantial portions of the Software.
-//
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
-// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-// THE SOFTWARE.
-//
-//------------------------------------------------------------------------------
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
 
 using Newtonsoft.Json;
-using System.Collections.Generic;
 
 namespace Microsoft.Identity.Test.LabInfrastructure
 {
     public class LabResponse
     {
-        [JsonProperty("AppId")]
-        public string AppId { get; set; }
+        [JsonProperty("app")]
+        public LabApp App { get; set; }
 
-        [JsonProperty("B2CAuthorities")]
-        public IList<B2CAuthority> B2CAuthoritities { get; set; }
+        [JsonProperty("user")]
+        public LabUser User { get; set; }
 
-        [JsonProperty("Authority")]
-        public IList<string> Authority { get; set; }
+        [JsonProperty("lab")]
+        public Lab Lab { get; set; }
+    }
 
-        [JsonProperty("Scope")]
-        public IList<string> Scopes { get; set; }
+    public class LabApp
+    {
+        [JsonProperty("appid")]
+        public string AppId { get; set; }
 
-        [JsonProperty("Users")]
-        public LabUser User { get; set; }
+        // TODO: this is a list, but lab sends a string. Not used today, discuss with lab to return a list
```

Comment:
I think they fixed this.

Reply:
we have the same comment in msal. i'll check on this.

Reply:
looks to be a string still. and they are heads down on security stuff. i'll follow up next week.

---
In reply to: [355231326](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1681#discussion_r355231326) [](ancestors = 355231326)

Reply:
Thx for following up. I think they have a work item on this already.